### ISSUE TITLE
Batch-gradient decent

### ISSUE THREAD

**Issue by Sibasish-Padhy** [Algo-Phantoms/Algo-ScriptML#139]
A very helpful optimization technique that is quite time efficient and even reduces space complexity. It proves to come in quite handy when the gradient descent  seems to have a relatively fast learning rate or follows non convex optimization. I would like to code the optimization technique and prepare a well documented readme file for it explaining it in detail. Please assign this issue to me under the label of  gssoc21.Thank you

**Comment by Sibasish-Padhy** [comment]
@ashwani-rathee can you please assign this issue to me under the label of gssoc 21

**Comment by Sibasish-Padhy** [comment]
thank you sir will try to make a pr as soon as possible but require more days sir.it is a request

**Comment by ashwani-rathee** [comment]
Sure @Sibasish-Padhy

**Comment by einhawking** [comment]
I am a participant GSSOC'21
I would like to contribute in this

**Comment by ashwani-rathee** [comment]
No progress on this,needs to be reassigned

**Comment by Sibasish-Padhy** [comment]
@ashwani-rathee  sir I am working on it my laptop was damaged needed to be repaired .hence it took a week.please sir understand the situation i will try to make a pr by  monday if possible

**Comment by Sibasish-Padhy** [comment]
please reassign me this issue sir @ashwani-rathee sir

**Comment by Sibasish-Padhy** [comment]
I have already started working on it

**Comment by ashwani-rathee** [comment]
@Sibasish-Padhy sure, Go ahead

**Comment by Sibasish-Padhy** [comment]
sir i have created a pull request for issue no #139.please do review it and suggest for changes. @ashwani-rathee

**Comment by subhangi2731** [comment]
@ashwani-rathee please assign  me this issue

**Comment by subhangi2731** [comment]
@ashwani-rathee ?? please assign

**Comment by Manasi2001** [comment]
Can I work on README for this @ashwani-rathee @geekquad @yukti845?
